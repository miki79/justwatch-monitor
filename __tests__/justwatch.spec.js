const nock = require('nock');
const fs = require('fs');
const { getNotification } = require('../src/justwatch');

console.info = jest.fn();

describe('Get Notifications from Justwatch', () => {
  afterEach(() => {
    nock.cleanAll();
  });
  test('Get 1 notification', () => {
    const token = '1234';
    nock('https://apis.justwatch.com')
      .get('/content/titles/en_GB/new/single_provider')
      .query(actualQueryObject => !!actualQueryObject.body.match(/nfx/))
      .reply(200, fs.readFileSync(`${__dirname}/stub/nfx.json`));
    nock('https://apis.justwatch.com')
      .get('/content/titles/en_GB/new/single_provider')
      .query(actualQueryObject => !!actualQueryObject.body.match(/amz/))
      .reply(200, fs.readFileSync(`${__dirname}/stub/amz-large.json`));
    nock('https://apis.justwatch.com')
      .get('/content/titles/en_GB/new/single_provider')
      .query(actualQueryObject => !!actualQueryObject.body.match(/amz/))
      .reply(200, fs.readFileSync(`${__dirname}/stub/amz-large.json`));
    nock('https://apis.justwatch.com')
      .get('/content/titles/en_GB/new/single_provider')
      .query(actualQueryObject => !!actualQueryObject.body.match(/skg/))
      .reply(200, fs.readFileSync(`${__dirname}/stub/skg.json`));
    nock('https://userapi.justwatch.com')
      .get('/store')
      .reply(200, fs.readFileSync(`${__dirname}/stub/one-watchlist.json`));
    return getNotification(token).then(notifications => expect(notifications).toMatchSnapshot());
  });
  test('Get no notification', () => {
    const token = '1234';
    nock('https://apis.justwatch.com')
      .get('/content/titles/en_GB/new/single_provider')
      .query(actualQueryObject => !!actualQueryObject.body.match(/nfx/))
      .reply(200, fs.readFileSync(`${__dirname}/stub/nfx.json`));
    nock('https://apis.justwatch.com')
      .get('/content/titles/en_GB/new/single_provider')
      .query(actualQueryObject => !!actualQueryObject.body.match(/amz/))
      .reply(200, fs.readFileSync(`${__dirname}/stub/amz.json`));
    nock('https://apis.justwatch.com')
      .get('/content/titles/en_GB/new/single_provider')
      .query(actualQueryObject => !!actualQueryObject.body.match(/skg/))
      .reply(200, fs.readFileSync(`${__dirname}/stub/skg.json`));
    nock('https://userapi.justwatch.com')
      .get('/store')
      .reply(200, fs.readFileSync(`${__dirname}/stub/no-watchlist.json`));
    return getNotification(token).then(notifications => expect(notifications).toMatchSnapshot());
  });
  test('Get 3 notifications', () => {
    const token = '1234';
    nock('https://apis.justwatch.com')
      .get('/content/titles/en_GB/new/single_provider')
      .query(actualQueryObject => !!actualQueryObject.body.match(/nfx/))
      .reply(200, fs.readFileSync(`${__dirname}/stub/nfx.json`));
    nock('https://apis.justwatch.com')
      .get('/content/titles/en_GB/new/single_provider')
      .query(actualQueryObject => !!actualQueryObject.body.match(/amz/))
      .reply(200, fs.readFileSync(`${__dirname}/stub/amz.json`));
    nock('https://apis.justwatch.com')
      .get('/content/titles/en_GB/new/single_provider')
      .query(actualQueryObject => !!actualQueryObject.body.match(/skg/))
      .reply(200, fs.readFileSync(`${__dirname}/stub/skg.json`));
    nock('https://userapi.justwatch.com')
      .get('/store')
      .reply(200, fs.readFileSync(`${__dirname}/stub/three-watchlist.json`));
    return getNotification(token).then(notifications => expect(notifications).toMatchSnapshot());
  });
  test('Get Error watchlist', () => {
    const token = '1234';
    nock('https://apis.justwatch.com')
      .get('/content/titles/en_GB/new/single_provider')
      .query(actualQueryObject => !!actualQueryObject.body.match(/nfx/))
      .reply(200, fs.readFileSync(`${__dirname}/stub/nfx.json`));
    nock('https://apis.justwatch.com')
      .get('/content/titles/en_GB/new/single_provider')
      .query(actualQueryObject => !!actualQueryObject.body.match(/amz/))
      .reply(200, fs.readFileSync(`${__dirname}/stub/amz.json`));
    nock('https://apis.justwatch.com')
      .get('/content/titles/en_GB/new/single_provider')
      .query(actualQueryObject => !!actualQueryObject.body.match(/skg/))
      .reply(200, fs.readFileSync(`${__dirname}/stub/skg.json`));
    nock('https://userapi.justwatch.com')
      .get('/store')
      .reply(500, '');
    return getNotification(token).catch(error =>
      expect(error.message).toBe('No body from Watchlist'));
  });
  test('Get Error new release', () => {
    const token = '1234';
    nock('https://apis.justwatch.com')
      .get('/content/titles/en_GB/new/single_provider')
      .query(actualQueryObject => !!actualQueryObject.body.match(/nfx/))
      .reply(200, fs.readFileSync(`${__dirname}/stub/nfx.json`));
    nock('https://apis.justwatch.com')
      .get('/content/titles/en_GB/new/single_provider')
      .query(actualQueryObject => !!actualQueryObject.body.match(/amz/))
      .reply(200, fs.readFileSync(`${__dirname}/stub/amz.json`));
    nock('https://apis.justwatch.com')
      .get('/content/titles/en_GB/new/single_provider')
      .query(actualQueryObject => !!actualQueryObject.body.match(/skg/))
      .reply(200, fs.readFileSync(`${__dirname}/stub/skg.json`));
    nock('https://userapi.justwatch.com')
      .get('/store')
      .reply(200, fs.readFileSync(`${__dirname}/stub/three-watchlist.json`));
    return getNotification(token).catch(error =>
      expect(error.message).toBe('No body from new release'));
  });
});
