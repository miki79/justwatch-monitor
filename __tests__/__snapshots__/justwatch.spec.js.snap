// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Notifications from Justwatch Get 1 notification 1`] = `
Array [
  Object {
    "object_type": "movie",
    "provider": "Netflix",
    "season": null,
    "title": "The Baby-Sitters Club",
  },
]
`;

exports[`Get Notifications from Justwatch Get 3 notifications 1`] = `
Array [
  Object {
    "object_type": "movie",
    "provider": "Netflix",
    "season": null,
    "title": "Pay the Ghost",
  },
  Object {
    "object_type": "movie",
    "provider": "Sky GO",
    "season": null,
    "title": "King Arthur: Legend of the Sword",
  },
]
`;

exports[`Get Notifications from Justwatch Get no notification 1`] = `Array []`;
